test/core.js

```diff
@@ -1134,6 +1134,57 @@ describe('Core', () => {
             expect(res.result).to.be.true();
         });
 
+        it('sets `request.isInjected = true` for requests created via `server.inject`', async () => {
+
+            const server = Hapi.server();
+            server.route({ method: 'GET', path: '/', handler: (request) => request.isInjected });
+
+            const options = {
+                url: '/'
+            };
+
+            const res = await server.inject(options);
+            expect(res.statusCode).to.equal(200);
+            expect(res.result).to.be.true();
+        });
+
+        it('`request.isInjected` access is read-only', async () => {
+
+            const server = Hapi.server();
+            server.route({ method: 'GET', path: '/', handler: (request) => {
+
+                try {
+                    request.isInjected = false;
+                }
+                catch (e) {
+                    // silence error to keep unit test output clean.  We expect above line to fail because isInjected() is getter type
+                }
```

Comment:
Can you use `expect(...).to.throw(...)` here.

Reply:
Implemented here https://github.com/hapijs/hapi/pull/4117/commits/0a99e50fec8a57c32df06b87a5ac84b84132fb8b, but then I rewrote it to https://github.com/hapijs/hapi/pull/4117/commits/674f3becf0eb14954381876fbc820f2ab47ec53a so that when unit test would fail, the failure message would be direct / obvious (In first commit, if unit test failed it would be that a 500 was received instead of a 200, but not immediately obvious to why that was)